File: TP2/Exo 2.4/python/copie.py
def copie(source):
    # Copie de la chaîne source dans une nouvelle chaîne
    destination = ''  # Initialisation d'une chaîne vide pour la copie
    for char in source:
        destination += char  # Ajoute chaque caractère de la source à la destination
    return destination

def main():
    # Demande à l'utilisateur de saisir une chaîne de caractères
    chaine = input("Saisissez une chaîne de caractères à copier : ")
    
    # Copie la chaîne en utilisant la fonction copie()
    copie_chaine = copie(chaine)
    
    # Affiche la copie de la chaîne
    print(f"Copie de la chaîne : {copie_chaine}")

# Point d'entrée du programme
if __name__ == "__main__":
    main()

File: TP2/Exo 2.4/C/copie.c
#include <stdio.h>

// Fonction qui copie la chaîne source dans la chaîne destination
void copie(char destination[], char source[])
{
    int i = 0;
    // Parcourt chaque caractère de la source jusqu'au caractère nul
    while (source[i] != '\0')
    {
        destination[i] = source[i]; // Copie chaque caractère
        i++;
    }
    destination[i] = '\0'; // Ajoute le caractère nul à la fin de la destination
}

int main()
{
    char source[100];      // Allouer suffisamment de place pour la chaîne source
    char destination[100]; // Allouer suffisamment de place pour la chaîne destination

    printf("Envoie la chaîne source : ");
    scanf("%99s", source); // Lire une chaîne de caractères jusqu'à 99 caractères

    copie(destination, source); // Copier la chaîne source dans la destination

    printf("La chaîne copiée est : %s\n", destination); // Afficher la chaîne copiée
    
    return 0;
}
